netbox/extras/models/customfields.py

```diff
@@ -215,7 +215,7 @@ def remove_stale_data(self, content_types):
         """
         for ct in content_types:
             model = ct.model_class()
-            instances = model.objects.filter(**{f'custom_field_data__{self.name}__isnull': False})
+            instances = model.objects.exclude(**{f'custom_field_data__contains': self.name})
```

Comment:
I think we want to look up by the presence of the field's name as a key.
```suggestion
            instances = model.objects.filter(custom_field_data__has_key=self.name)
```

Reply:
@jeremystretch  yeah - that would be better. fixed.